src/stylesheets/components/_article.scss

```diff
@@ -44,4 +44,41 @@ article {
       }
     }
   }
+
+  & + aside .block .list--pipe:last-child > li:last-child {
+    border-bottom: 0;
+  }
+
+  @media (min-width: $tablet) {
+
+    // These styles for `h2, p, ol, ul` tags are to align
+    // the baseline of each of those element's text with
+    // the baseline of the `h2` tag inside the `article`
+    // tag's ajacent sibling the `aside` tag.
+    //
+    // This also fixes the extra spacing of 18px a
+    // first-child 'h2' tag makes when inside a new
+    // `section` tag. In this scenario, this fix is only
+    // happening when the `h2` tag is inside the `article`
+    // tag. However, there is now extra space above the
+    // first-child `p, ol, ul` tags to align their
+    // first-line text baselines with the `h2` tag inside
+    // the `aside` tag.
+
+    h2:first-child {
+      margin-top: $base-spacing-unit-medium;
+    }
+    p:first-child,
+    > ol:first-child > li:first-child,
```

Comment:
I know not part of this PR and it is an extension of what already has been done but is not great if the child element margins go beyond the parent. As a general pattern also no component or element except headings have a top margin. Example is `.list--block`